packages/rsc-builder/buildRsc.ts

```diff
@@ -27,6 +27,7 @@ const PROJECTS: Project[] = [
       'packages/mui-material/src/PigmentContainer/PigmentContainer.tsx', // RSC compatible
       'packages/mui-material/src/PigmentGrid/PigmentGrid.tsx', // RSC compatible
       'packages/mui-material/src/PigmentStack/PigmentStack.tsx', // RSC compatible
+      'packages/mui-material/src/useAutocomplete/useAutocomplete.js', // RSC compatible
```

Comment:
I agree, we should remove https://github.com/mui/material-ui/blob/c1c0d7c73fdb585d468ea106f8ef31a65bf6e49a/packages/mui-material/src/useAutocomplete/useAutocomplete.js#L1

From a quick test that I could don't with Next.js. If we add the use client in the hook, we get no clear error when used wrong. Next.js just gives you an empty import on the server bundle when trying to import from the client bundle.

Reply:
* Next.js only starts erroring on this once we add the `exports` field to package.json. I have the feeling it's not as strict with commonjs. That's why you currently can't reproduce.
* We have a [lint rule](https://github.com/mui/material-ui/blob/c1c0d7c73fdb585d468ea106f8ef31a65bf6e49a/.eslintrc.js#L226-L230). It checks whether `'use client'` and `export *` are used in the same file. This is the pattern that Next.js doesn't like and thus we should avoid. The fact that we had a lot of those in index files is a coincidence.
* imo buildRsc is not the optimal approach to this problem. We should just add `'use client'` where it's necessary.  I think it would be better to write a test instead that imports all our components in an RSC so that we don't forget this. Just in general, our preference for solving these sorts of problems should be linting/testing first, before code generation.

Reply:
@Janpot Ah no, I was talking about another type of error. In Next.js, when I import a hook from a server component, the DX changes depending on either the hook has use client or not.

1. if the hook has "use client": it imports an empty object `{}`. When trying to call it, it crashes with no clear reason why.
2. if the hook doesn't have "use client": Next.js complains that you are trying to use a React client side API without use client, it asks you to add one

2. looks clearly better to me;

Reply:
My bad, my previous message don't holp up when looking at it in more depth: https://github.com/mui/material-ui/pull/43778. I only considered Next.js@latest but they had a regression in the DX, they will fix it.